client/state/stats/lists/reducer.js

```diff
@@ -73,17 +73,14 @@ export function items( state = {}, action ) {
 		case SITE_STATS_RECEIVE:
 			const queryKey = getSerializedStatsQuery( action.query );
 
-			// To avoid corrupted stat data with massive arrays
-			// From causing _.merge to crash, first clone, then unset existing data
-			const existingItems = Object.assign( {}, state );
-			unset( existingItems, [ action.siteId, action.statType, queryKey ] );
-			return merge( {}, existingItems, {
-				[ action.siteId ]: {
+			return {
+				...state,
+				[ action.siteId ]: merge( {}, state[ action.siteId ], {
 					[ action.statType ]: {
 						[ queryKey ]: action.data
```

Comment:
I'm not sure using `merge` here is suited. Because if we have the same `statType` and `queryKey`. The previous `action.data` will be merged with the new `action.data` and since we don't really know the shape of this object here. We'll probably want to replace it instead of merging it with the previous one.